addon/lazy-router.js

```diff
@@ -35,12 +35,18 @@ export default Ember.Router.extend({
         }
       }
 
-      handler.routeName = name;
+      if (typeof handler._setRouteName === 'function') {
+        handler._setRouteName(name);
+        handler._populateQPMeta();
+      } else {
+        handler.routeName = name;
+      }
       return handler;
     };
   },
 
-  _queryParamsFor: function(leafRouteName) {
+  _queryParamsFor: function(handlerInfos) {
```

Comment:
nit: we could rename the param to reflect the new signature difference. Right now it's handlerInfosOrLeafRouteName.